src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1347,6 +1399,17 @@ void BlockchainLMDB::open(const std::string& filename, const int db_flags)
     throw DB_ERROR("Database could not be opened");
   }
 
+#ifdef WIN32
+  // ensure NTFS compression is disabled on the directory and database file to avoid corruption of the blockchain 
+  if (!disable_ntfs_compression(filename)) 
+    LOG_PRINT_L0("Failed to disable NTFS compression on folder: " << filename << ". Error: " << ::GetLastError());
+  boost::filesystem::path datafile(filename);
+  datafile /= CRYPTONOTE_BLOCKCHAINDATA_FILENAME;
+  boost::filesystem::ofstream(datafile.string()).close(); // touch the file to ensure it exists
```

Comment:
I think you can simplify this to `ofstream(datafile)`, which avoids information loss from path->string conversion

Reply:
Fixed.